### PR TITLE
Add auto github release workflow

### PR DESCRIPTION
**Pull request by reta**
### Description
Add auto Github release workflow

### Issues Resolved
Closes https://github.com/opensearch-project/security/issues/2435

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/auto-release.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: GitHub App token
+        id: github_app_token
+        uses: tibdex/github-app-token@v1.5.0
+        with:
+          app_id: ${{ secrets.APP_ID }}
```
Comment (reta) [inline .github/workflows/auto-release.yml:18]: @saratvemulapalli do we need to configure APP_ID & APP_PRIVATE_KEY for this repository individually? thanks!

Comment (cwperks) [reply .github/workflows/auto-release.yml:18]: @reta APP_ID and APP_PRIVATE_KEY are already configured for this repo in the repo settings.

Comment (cwperks) [reply .github/workflows/auto-release.yml:18]: I was able to test this on my fork of security-dashboards-plugin by creating a new github app and installing it on my fork

Comment (reta) [reply .github/workflows/auto-release.yml:18]: Thanks @cwperks !

**Review by cwperks — APPROVED** [review]

**Review by RyanL1997 — APPROVED** [review]

**Review by scrawfor99 — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
@reta this has to be backported to release branches if not the workflow doesnt run when a tag is pushed to that branch.

**Comment by cwperks** [comment]
@saratvemulapalli Added backport label to this and corresponding PR in the security-dashboards-plugin repo

**Pull request merged**